aten/src/TH/generic/THStorage.cpp

```diff
@@ -137,25 +137,34 @@ void THStorage_(resize)(THStorage *storage, ptrdiff_t size)
 void THStorage_(fill)(THStorage *storage, real value)
 {
   ptrdiff_t i;
-  for(i = 0; i < storage->size; i++)
+  for(i = 0; i < storage->size(); i++)
     THStorage_(data)(storage)[i] = value;
 }
 
 void THStorage_(set)(THStorage *self, ptrdiff_t idx, real value)
 {
-  THArgCheck((idx >= 0) && (idx < self->size), 2, "out of bounds");
+  THArgCheck((idx >= 0) && (idx < self->size()), 2, "out of bounds");
   THStorage_(data)(self)[idx] = value;
 }
 
 real THStorage_(get)(const THStorage *self, ptrdiff_t idx)
 {
-  THArgCheck((idx >= 0) && (idx < self->size), 2, "out of bounds");
+  THArgCheck((idx >= 0) && (idx < self->size()), 2, "out of bounds");
   return THStorage_(data)(self)[idx];
 }
 
 void THStorage_(swap)(THStorage *storage1, THStorage *storage2)
 {
-  THStorage_swap(storage1, storage2);
+  std::swap(storage1->scalar_type(), storage2->scalar_type());
+  std::swap(storage1->data_ptr(), storage2->data_ptr());
+  ptrdiff_t tmp_size = storage1->size();
+  storage1->set_size(storage2->size());
+  storage2->set_size(tmp_size);
+  bool tmp_bool = storage1->resizable();
+  storage1->set_resizable(storage2->resizable());
+  storage2->set_resizable(tmp_bool);
+  std::swap(storage1->allocator_, storage2->allocator_);
+  std::swap(storage1->finalizer_, storage2->finalizer_);
```

Comment:
I know I gave conflicting advice here, but I think my preferred implementations of swap is using a hypothetical move constructor on storage. The move constructor is closer to the fields and more likely to be updated if someone adds another field. But this is OK for now.

Reply:
Makes sense. I was planning on doing something like that in an upcoming PR as this is becoming too large.

Reply:
This function is now screaming to be implemented as a method (or, at very least, a friend function.)

```
  bool tmp_bool = storage1->resizable();
  storage1->set_resizable(storage2->resizable());
  storage2->set_resizable(tmp_bool);
```

is terrible and we shouldn't encourage it.